packages/lib/src/vis/shared/AxisSystemProvider.tsx

```diff
@@ -46,6 +54,20 @@ function AxisSystemProvider(props: PropsWithChildren<Props>) {
     [camera, cameraToHtmlMatrix]
   );
 
+  const shouldInteract = useCallback(
+    (id: string, event: MouseEvent | WheelEvent) => {
+      if (!Object.keys(interactionKeys).includes(id)) {
+        throw new Error(`Interaction ${id} was not registered in VisCanvas.`);
+      }
+      const modifierKeyPressed = Object.values(interactionKeys).find(
+        (key) => !!key && event.getModifierState(key)
+      );
+
+      return modifierKeyPressed === interactionKeys[id];
```

Comment:
Although I get why this works, I feel like it would be clearer to treat the two cases (whether `interactionKeys[id]` is defined or not) separately: when defined, you just return `event.getModifierKey(interactionKeys[id])`; otherwise, you return `Object.values(interactionKeys).find(...)`. Makes sense?

To deal with the case where two interactions are bound to the same key, you could, before the `useCallback`, initialise an array of all the modifier keys in the `interactionKeys` object. Then, you put that in a `Set` and compare the length of the array with the size of the set and throw if they're different. Then, you can loop through the set where you need to inside the `useCallback`.

When it comes to pressing multiple keys, I don't see it as a problem - e.g. `Alt + Shit + Wheel` is just equivalent to normal zoom. If you put the normal zoom on `Alt` and keep the Y-axis zoom on `Shift`, then you just do a normal zoom (I think?) I guess it becomes a bit more problematic with drag interactions? Not sure. Either way, I'm more for waiting to see if a problematic combination comes up.